app/controllers/tasks_controller.rb

```diff
@@ -2,22 +2,32 @@ class TasksController < ApplicationController
   before_action :set_task, only: [:show, :edit, :update, :destroy]
   before_action :set_status, only: [:index,:create, :new, :edit, :update, :destroy]
   def index
-    @tasks = Task.all
+    redirect_to statuses_path
   end
 
   def show
   end
 
   def new
-    @task = Task.new
+    @user = current_user
+  @status = Status.find_by(id: params[:status_id])
```

Comment:
Add this code into before_action as this is getting repeated multiple times in other methods as well.